pkg/labels/consul_aggregator.go

```diff
@@ -184,7 +154,56 @@ func (c *consulAggregator) Aggregate() {
 		case <-c.aggregatorQuit:
 			return
 		case <-loopTime:
+			// we purposely don't case outErrors here, since loopTime lets us
+			// back off of Consul watches. If an error repeatedly were occurring,
+			// we could end up in a nasty busy loop.
+		}
+	}
+}
+
+func (c *consulAggregator) fillCache(pairs api.KVPairs) {
+	cache := make([]Labeled, len(pairs))
+	c.labeledCache = cache
```

Comment:
it seems weird to set this before we've populated any values. I think the behavior is correct because this is done while holding the locks but maybe do this after?